genesyscloud/resource_genesyscloud_externalcontacts_contact.go

```diff
@@ -0,0 +1,812 @@
+package genesyscloud
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"terraform-provider-genesyscloud/genesyscloud/consistency_checker"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mypurecloud/platform-client-sdk-go/v92/platformclientv2"
+)
+
+var (
+	phoneNumber = &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"display": {
+				Description: "Display string of the phone number.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"extension": {
+				Description: "Phone extension.",
+				Type:        schema.TypeInt,
+				Optional:    true,
+			},
+			"accepts_sms": {
+				Description: "If contact accept SMS.",
+				Type:        schema.TypeBool,
+				Optional:    true,
+			},
+			"e164": {
```

Comment:
I would add a ValidateDiagFunc: validatePhoneNumber function to this resource block

Reply:
added in [this commit](https://github.com/MyPureCloud/terraform-provider-genesyscloud/pull/466/commits/7d3af16150df89e25b08a872e713219c3e7164a9)